pkg/terraform/terraform.go

```diff
@@ -53,24 +54,49 @@ func (m *Mixin) getPayloadData() ([]byte, error) {
 }
 
 func (m *Mixin) getOutput(outputName string) ([]byte, error) {
+	// Using -json instead of -raw because terraform only allows for string, bool,
+	// and number output types when using -raw. This means that the outputs will
+	// need to be unencoded to raw string because -json does json compliant html
+	// special character encoding.
 	cmd := m.NewCommand("terraform", "output", "-json", outputName)
 	cmd.Stderr = m.Err
 
 	// Terraform appears to auto-append a newline character when printing outputs
 	// Trim this character before returning the output
 	out, err := cmd.Output()
-	out = bytes.TrimRight(out, "\n")
-	// Terraform quotes simple object types when using -json format argument
-	// No object type should be quoted at the top-level so trim those quotes if they exist
-	out = bytes.TrimLeft(out, "\"")
-	out = bytes.TrimRight(out, "\"")
-
 	if err != nil {
 		prettyCmd := fmt.Sprintf("%s %s", cmd.Path, strings.Join(cmd.Args, " "))
 		return nil, errors.Wrap(err, fmt.Sprintf("couldn't run command %s", prettyCmd))
 	}
 
-	return out, nil
+	// Implement a custom JSON encoder that doesn't do HTML escaping. This allows
+	// for recrusive decoding of complex JSON objects using the unmarshal and then
+	// re-encoding it but skipping the html escaping. This allows for complex types
+	// like maps to be represented as a byte slice without having go types be
+	// part of that byte slice, eg: without the re-encoding, a JSON byte slice with
+	// '{"foo": "bar"}' would become map[foo:bar].
+	var outDat interface{}
+	err = json.Unmarshal(out, &outDat)
+	if err != nil {
+		return []byte{}, err
+	}
+	buffer := &bytes.Buffer{}
+	encoder := json.NewEncoder(buffer)
+	encoder.SetEscapeHTML(false)
+	err = encoder.Encode(outDat)
+	if err != nil {
+		return []byte{}, err
+	}
+	outJSON := bytes.TrimRight(buffer.Bytes(), "\n")
+	// If the output data is of type string then do one last unmarshal to json
+	// string so that the outer quotes are stripped, otherwise return the raw
+	// json syntax directly.
+	var outString string
+	if err = json.Unmarshal(outJSON, &outString); err == nil {
```

Comment:
Do we still need to unmarshal the entire output json in order to remove the quotes here?

Reply:
I could also do type assertion and skip then re-encode for string types then we wouldn't need this second unmarshal